packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -26,9 +26,9 @@ export function makeCache(size, fetch, store) {
   const cache = {
     makeRoom() {
       while (liveTable.size > size && lruTail) {
-        if (lruTail.rawData[initializationInProgress]) {
+        if (initializationsInProgress.has(lruTail.rawData)) {
```

Comment:
I never understood what race condition this protects against. I'm reviewing ignoring that question.

Reply:
I can explain it in detail if you care, but tl;dr: async calls from `init` can potentially get you into a situation where you fill the entire cache with objects being initialized and it tries to page out an object whose initialization is still in progress and thus has no valid state object to swap out.

Reply:
That makes sense. Thanks.